src/include/packager/repositories_include.rb

```diff
@@ -469,5 +327,185 @@ def autorefresh_for?(url)
       log.info "Autorefresh flag for '#{protocol}' URL protocol: #{autorefresh}"
       autorefresh
     end
+
+  private
+
+    # scan the repository URL and return the available products
+    # @return [Array<Y2Packager::ProductLocation>] Found products
+    def scan_products(expanded_url, original_url)
+      new_repos = Pkg.RepositoryScan(expanded_url)
+      found_products = new_repos.map { |r| Y2Packager::ProductLocation.new(r[0], r[1]) }
```

Comment:
`found_products = new_repos.map { |(name, dir)| Y2Packager::ProductLocation.new(name, dir) }` is more readable.